tests/e2e/configurer/setup_runner.go

```diff
@@ -0,0 +1,62 @@
+package configurer
+
+import (
+	"fmt"
+)
+
+type setupFn func(configurer Configurer) error
+
+func baseSetup(configurer Configurer) error {
```

Comment:
This file uses a function decorator pattern because depending on what is enabled (IBC, upgrade or state sync testing), we want to run a different kind of setup logic

This is how we decide which decorators to run: https://github.com/osmosis-labs/osmosis/blob/c5d5c9f0c6b5c7fdf9688057eb78ec793f6dd580/tests/e2e/configurer/configurer.go#L166